### PR TITLE
DAOS-623 build: Enable avx2 in build

### PR DESCRIPTION
**Pull request by jolivier23**
The DESIRED_FLAGS entry for -mavx2 had a space which causes some
random behavior.  On my box and on boro, it causes it to fail the
configure check even if the option is available.  Neil tried
building on his OpenSuse laptop and it passed the configure
check but then failed to compile anything because it passed
the option as " -mavx2" rather than just -mavx2.

Removing the space fixes it and probably enables the extention
on most systems.

Signed-off-by: Jeff Olivier <jeffrey.v.olivier@intel.com>

### CODE REVIEW

**Review by mchaarawi — APPROVED** [review]

**Comment by jolivier23** [comment]
@johannlombardi where did you get this implementation?   There is no d_cmp16 defined so it doesn't compile.  Actually, it does seem to compile on some platforms but not sure why.

**Comment by jolivier23** [comment]
Probably don't want to merge this as is...Need a real fix but this should at least compile now.

File: src/include/gurt/mem.h [hunk 1]
```diff
@@ -20,7 +20,7 @@
 extern "C" {
 #endif
 
-#include <string.h>
+#include <gurt/common.h>
```
Comment (jolivier23) [inline src/include/gurt/mem.h:23]: I guess the issue in proc is due to debug stuff getting initialized?   That's the problem with common.h, it adds too much crap on top of what is absolutely necessary.  CART-596 and DAOS-6251 touch on the issues

**Comment by jolivier23** [comment]
It doesn't appear to work.  Maybe we should land a patch that just removes -mavx2 altogether from DESIRED_FLAGS and work on this in the background ?

**Comment by johannlombardi** [comment]
Agree, let's keep avx2 since i think that we should use it regardless and just disable the optimized memcpy/cmp. I will fix it in the background. Thanks.

**Pull request merged**